lucid/misc/io/loading.py

```diff
@@ -187,14 +224,22 @@ def load(url_or_handle, cache=None, **kwargs):
 # Helpers
 
 
-def load_using_loader(url_or_handle, loader, cache, **kwargs):
+def load_using_loader(url_or_handle, decompressor, loader, cache, **kwargs):
     if is_handle(url_or_handle):
-        result = loader(url_or_handle, **kwargs)
+        if decompressor:
+            with decompressor(url_or_handle) as decompressor_handle:
+                result = loader(decompressor_handle, **kwargs)
+        else:
+            result = loader(url_or_handle, **kwargs)
     else:
         url = url_or_handle
         try:
             with read_handle(url, cache=cache) as handle:
-                result = loader(handle, **kwargs)
+                if decompressor:
+                    with decompressor(handle) as decompressor_handle:
+                        result = loader(decompressor_handle, **kwargs)
+                else:
+                    result = loader(handle, **kwargs)
```

Comment:
(nitpicking, feel free to disregard) The logic here feels like it's getting a bit deep, and also repeated. I wonder if you could do something like the following to simplify things:

```py
if decompressor is None:
   decompressor = contextlib.nullcontext
```

and drop all further logic about whether something is being decompressed.